biojava-core/src/main/java/org/biojava/nbio/core/search/io/blast/BlastXMLQuery.java

```diff
@@ -0,0 +1,270 @@
+package org.biojava.nbio.core.search.io.blast;
+
+
+import org.biojava.nbio.core.search.io.Hit;
+import org.biojava.nbio.core.search.io.Hsp;
+import org.biojava.nbio.core.search.io.Result;
+import org.biojava.nbio.core.search.io.ResultFactory;
+import java.io.File;
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Logger;
+import javax.xml.xpath.XPathException;
+import org.biojava.nbio.core.sequence.template.Sequence;
+import org.biojava.nbio.core.util.XMLHelper;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+
+/**
+ * Re-designed by Paolo Pavan.
+ * You may want to find my contacts on Github and LinkedIn for code info 
+ * or discuss major changes.
+ * https://github.com/paolopavan
+ * 
+ *
+ * @author Scooter Willis <willishf at gmail dot com>
+ * @author Paolo Pavan
+ */
+public class BlastXMLQuery implements ResultFactory {
+    private static final Logger log = Logger.getLogger(BlastXMLQuery.class.getName());
+    Document blastDoc = null;
+    private File targetFile;
+    private List<Sequence> queryReferences, databaseReferences;
+    private Map<String,Sequence> queryReferencesMap, databaseReferencesMap;
+    
+    @Deprecated
+    public BlastXMLQuery(String blastFile) throws Exception {
+        readFile(blastFile);
+    }
+    
+    public BlastXMLQuery() {
+        
+    }
+    public void setFile(File f){
+        targetFile = f;
+    }
+    /**
+     * 
+     * @param maxEScore
+     * @return
+     * @throws Exception 
+     */
+    @Deprecated
+    public LinkedHashMap<String, ArrayList<String>> getHitsQueryDef(double maxEScore) throws Exception {
+        LinkedHashMap<String, ArrayList<String>> hitsHashMap = new LinkedHashMap<String, ArrayList<String>>();

```

Comment:
Alright in that case I think it is a good idea to move it to SearchIO (removing the old one `org.biojava.nbio.genome.query.BlastXMLQuery`) and deprecating the methods as you did. Since as you say it seems to be only a stub and it isn't used much. 

If you go that way, one thing is important: whenever we deprecate things we should not use the deprecated methods within Biojava anymore (their usages would show as warnings). Deprecations are there to let other users know that we are moving forward and that in next versions the methods will be gone.

Another possibility can be simply leave the `org.biojava.nbio.genome.query.BlastXMLQuery` class as it is and don't touch it. Then create your own XML parser class with a different name in `search.io` package. I think that might be cleaner after all.